### DISCUSSION TITLE
Cancel PlayerStartFlyingEvent

### DISCUSSION THREAD

**Discussion by jandrokt** [Minestom/Minestom#695] [Help]
The title is pretty self explanatory. 
I've been looking for a way to do it for some time and I haven't found any. There's no way to cancel the event as of that so I was wondering if it's even possible to do it. 
Using player.setFlying(false) results in a funky high jump so it's really not ideal.
Thanks in advance :p

**Comment by ZakShearman** [comment]
You should be able to not allow the player to fly in the first place (Player#setAllowFlying). Is this a situation where this isn't feasible?

**Reply by jandrokt** [reply]
Well I'm trying to create a double jump system so I don't think it's feasible.

**Reply by ZakShearman** [reply]
From what I can tell I believe the event could be made cancellable. I'd wait for someone who knows more to reply and then maybe make an issue based on what they say 😃

**Comment by MrBretze** [comment]
Player.setFlying(false) ? I believe it works for me ?

**Reply by jandrokt** [reply]
This leaves a weird "jump", as if you were flying for a very short moment, so it's really funky

**Comment by MrlnHi** [comment]
Since you said that you wanted to make a double jump system, can't you just do
```java
player.setFlying(false);
player.setVelocity(velocity);
```
in your event?

**Reply by MrlnHi** [reply]
If your set velocity doesn't do anything, try running it the next tick, as in:
```java
player.scheduleNextTick(() -> player.setVelocity(velocity));
```

**Comment by jandrokt** [comment]
Here's a video that shows the weird velocity: https://youtu.be/jH9bDzGosoc

**Reply by MrlnHi** [reply]
To me that doesn't look like velocity, it's just something like velocity reset, meaning the client stops falling because it flies, then the server tells it it doesn't fly anymore and the client starts falling again, having to accelerate from 0 again

**Reply by MrBretze** [reply]
You cannot fix this, it's a Minecraft Client/Server Sync issue, this is not possible to fix

**Comment by emortaldev** [comment]
Oh, when you said velocity I thought you meant quite high, you can't stop the client from flying for a second unfortunately, however if you're looking for a double jump, this doesn't really matter?
